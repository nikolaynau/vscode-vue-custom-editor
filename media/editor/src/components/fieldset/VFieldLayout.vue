<script setup lang="ts">
import { toRefs, computed } from 'vue';

export interface Props {
  vertical?: boolean;
}

const props = withDefaults(defineProps<Props>(), {
  vertical: false
});

const { vertical } = toRefs(props);

const cssClasses = computed(() => ({
  'v-field-layout--horizontal': !vertical.value,
  'v-field-layout--vertical': vertical.value
}));
</script>

<template>
  <div
    v-if="$slots.label || $slots.value"
    class="v-field-layout"
    :class="cssClasses"
  >
    <div v-if="$slots.label" class="v-field-layout__label">
      <slot name="label"></slot>
    </div>
    <div v-if="$slots.value" class="v-field-layout__value">
      <slot name="value"></slot>
    </div>
  </div>
</template>

<style>
@import 'field-layout';
</style>
