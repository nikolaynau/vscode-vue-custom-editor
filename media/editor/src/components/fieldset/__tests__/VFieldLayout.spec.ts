import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import VFieldLayout from '../VFieldLayout.vue';

describe('VFieldLayout', () => {
  it('default render props', () => {
    const wrapper = mount(VFieldLayout);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('label slot', () => {
    const wrapper = mount(VFieldLayout, { slots: { label: 'Some label' } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('value slot', () => {
    const wrapper = mount(VFieldLayout, { slots: { value: 'Input' } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('render label and value slots', () => {
    const wrapper = mount(VFieldLayout, {
      slots: { label: 'Label', value: 'Input' }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('vertical prop', async () => {
    const wrapper = mount(VFieldLayout, {
      props: { vertical: false },
      slots: { label: 'Label', value: 'Input' }
    });
    expect(wrapper.classes('v-field-layout--horizontal')).toBe(true);

    await wrapper.setProps({ vertical: true });

    expect(wrapper.classes('v-field-layout--vertical')).toBe(true);
  });
});
