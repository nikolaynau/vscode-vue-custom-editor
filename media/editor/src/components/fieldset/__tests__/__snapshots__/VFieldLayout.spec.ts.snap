// Vitest Snapshot v1

exports[`VFieldLayout > default render props 1`] = `"<!--v-if-->"`;

exports[`VFieldLayout > label slot 1`] = `
"<div class=\\"v-field-layout v-field-layout--horizontal\\">
  <div class=\\"v-field-layout__label\\">Some label</div>
  <!--v-if-->
</div>"
`;

exports[`VFieldLayout > render label and value slots 1`] = `
"<div class=\\"v-field-layout v-field-layout--horizontal\\">
  <div class=\\"v-field-layout__label\\">Label</div>
  <div class=\\"v-field-layout__value\\">Input</div>
</div>"
`;

exports[`VFieldLayout > value slot 1`] = `
"<div class=\\"v-field-layout v-field-layout--horizontal\\">
  <!--v-if-->
  <div class=\\"v-field-layout__value\\">Input</div>
</div>"
`;
